esti/lakectl_local_test.go

```diff
@@ -169,11 +186,99 @@ func TestLakectlLocal_clone(t *testing.T) {
+	tmpDir := t.TempDir()
+	_, err := os.CreateTemp(tmpDir, "")
+	require.NoError(t, err)
+	repoName := generateUniqueRepositoryName()
+	storage := generateUniqueStorageNamespace(repoName)
+	vars := map[string]string{
+		"REPO":    repoName,
+		"STORAGE": storage,
+		"BRANCH":  mainBranch,
+	}
+
+	RunCmdAndVerifyFailureWithFile(t, Lakectl()+" log lakefs://"+repoName+"/"+mainBranch, false, "lakectl_log_404", vars)
+	RunCmdAndVerifySuccessWithFile(t, Lakectl()+" repo create lakefs://"+repoName+" "+storage, false, "lakectl_repo_create", vars)
+	RunCmdAndVerifySuccessWithFile(t, Lakectl()+" log lakefs://"+repoName+"/"+mainBranch, false, "lakectl_log_initial", vars)
+
+	// Not linked
+	vars["LOCAL_DIR"] = tmpDir
+	RunCmdAndVerifyFailureWithFile(t, Lakectl()+" local pull "+tmpDir, false, "lakectl_local_no_index", vars)
+
+	tests := []struct {
+		name   string
+		prefix string
+	}{
+		{
+			name:   "root",
+			prefix: "",
+		},
+		{
+			name:   "prefix",
+			prefix: "images",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dataDir, err := os.MkdirTemp(tmpDir, "")
+			require.NoError(t, err)
+			vars["PREFIX"] = "/" + tt.prefix
+			vars["LOCAL_DIR"] = dataDir
+			vars["BRANCH"] = tt.name
+			vars["REF"] = tt.name
+			runCmd(t, Lakectl()+" branch create lakefs://"+repoName+"/"+vars["BRANCH"]+" --source lakefs://"+repoName+"/"+mainBranch, false, false, vars)
+			RunCmdAndVerifyContainsText(t, Lakectl()+" local clone lakefs://"+repoName+"/"+vars["BRANCH"]+vars["PREFIX"]+" "+dataDir, false, "Successfully cloned lakefs://${REPO}/${REF}${PREFIX} to ${LOCAL_DIR}.", vars)
+
+			// Pull nothing
+			vars["DOWNLOADED"] = "0"
+			vars["REMOVED"] = "0"
+			RunCmdAndVerifyContainsText(t, Lakectl()+" local pull "+dataDir, false, successStr, vars)
+
+			// Upload and commit an object
+			base := []string{
+				"ro_1k.1",
+				"ro_1k.2",
+				"ro_1k.3",
+				"images/1.png",
+				"images/2.png",
+				"images/3.png",
+			}
+
+			modified := []string{"ro_1k.1"}
+			deleted := []string{"deleted"}
+			create := append(base, deleted...)
+			createTestData(t, vars, create)
+			expected := getExpected(t, tt.prefix, create)
+			// Pull changes and verify data
+			vars["DOWNLOADED"] = strconv.Itoa(len(expected))
+			vars["REMOVED"] = "0"
+			RunCmdAndVerifyContainsText(t, Lakectl()+" local pull "+dataDir, false, successStr, vars)
+			verifyDirContents(t, dataDir, expected)
+
+			// Modify data
+			vars["FILE_PATH"] = modified[0]
+			runCmd(t, Lakectl()+" fs upload -s files/ro_1k_other lakefs://"+vars["REPO"]+"/"+vars["BRANCH"]+"/"+vars["FILE_PATH"], false, false, vars)
+			runCmd(t, Lakectl()+" fs rm lakefs://"+repoName+"/"+vars["BRANCH"]+"/"+deleted[0], false, false, vars)
+			runCmd(t, Lakectl()+" commit lakefs://"+vars["REPO"]+"/"+vars["BRANCH"]+" --allow-empty-message -m \" \"", false, false, vars)
+			expected = slices.DeleteFunc(expected, func(s string) bool {
+				return slices.Contains(deleted, s)
```

Comment:
The deleted is one item - it means we don't need a slice of `deleted` as the code also run a single command and reference `deleted[0]`.
So having when we delete the item from the slice we can just:
```suggestion
				return s == deleted
```